site/content/en/docs/Tasks/labels_and_annotations.md

```diff
@@ -2,7 +2,293 @@
+
+## Working with Labels
+### Add Labels
+[`labels`] can be used to add labels to the `metadata` field of all Resources in a project. This will override values for label keys that already exist.
+
+Here is an example of how to add labels to the `metadata` field.
+1. Create a Kustomization file.
+```yaml
+# kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+labels:
+- pairs:
+    someName: someValue
+    owner: alice
+    app: bingo
+
+resources:
+- deploy.yaml
+- service.yaml
+```
+
+2. Create Deployment and Service manifests.
+```yaml
+# deploy.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: example
+```
+```yaml
+# service.yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: example
+```
+
+3. Add labels with `kustomize build`.
+```bash
+kustomize build .
+```
+
+The output shows that the `labels` field is used to add labels to the `metadata` field of the Service and Deployment Resources.
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: bingo
+    owner: alice
+    someName: someValue
+  name: example
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: bingo
+    owner: alice
+    someName: someValue
+  name: example
+```
+### Add Template Labels
+[`labels.includeTemplates`] can be used to add labels to the `spec.template` field of all applicable Resources in a project.
+
+Here is an example of how to add labels to the `spec.template` field of a Deployment.
+1. Create a Kustomization file.
+```yaml
+# kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+labels:
+- pairs:
+    someName: someValue
+    owner: alice
+    app: bingo
+  includeTemplates: true
+
+resources:
+- deploy.yaml
+- service.yaml
+```
+
+2. Create Deployment and Service manifests.
+```yaml
+# deploy.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: example
+```
+```yaml
+# service.yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: example
+```
+
+3. Add labels with `kustomize build`.
+```bash
+kustomize build .
+```
+
+The output shows that the `labels.includeTemplates` field is used to add labels to the `spec.template` field of the Deployment. The Service Resource API does not have a `spec.template` field and Kustomize does not add this field.
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: bingo
+    owner: alice
+    someName: someValue
+  name: example
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: bingo
+    owner: alice
+    someName: someValue
+  name: example
+spec:
+  template:
+    metadata:
+      labels:
+        app: bingo
+        owner: alice
+        someName: someValue
+```
+### Add Selector Labels
+[`labels.includeSelectors`] can be used to add labels to the `spec.selectors` field of all Resources in a project. Note that this also adds labels to the `spec.template` field for applicable Resources.
```

Comment:
We want to
* specify "selector field" instead of "`spec.selectors` field" because of the same reason as templates commented above
* change "all Resources" to "all applicable Resources": this file shows object `kind`s that have selectors https://github.com/kubernetes-sigs/kustomize/blob/master/api/internal/konfig/builtinpluginconsts/commonlabels.go

Also, can we likewise change "`spec.template` field" here to "template field"?